packages/fuels-abigen-macro/tests/harness.rs

```diff
@@ -671,6 +672,38 @@ async fn type_safe_output_values() {
     assert!(response.value.bar);
 }
 
+#[tokio::test]
+async fn panics_on_non_binary_file() {
+    // Generates the bindings from the an ABI definition inline.
+    // The generated bindings can be accessed through `SimpleContract`.
+    abigen!(
+        MyContract,
+        "packages/fuels-abigen-macro/tests/test_projects/contract_output_test/out/debug/contract_output_test-abi.json"
+    );
+
+    let wallet = launch_provider_and_get_single_wallet().await;
+
+    println!("here");
+    // Should panic as we are passing in a JSON instead of BIN
+    let deploy_result = Contract::deploy(
+        "tests/test_projects/contract_output_test/out/debug/contract_output_test-abi.json",
+        &wallet,
+        TxParameters::default(),
+    )
+    .await;
+    assert!(deploy_result.is_err());
+
+    // Should panic as we are passing in a JSON instead of BIN
+    let deploy_result = Contract::deploy_with_salt(
+        "tests/test_projects/contract_output_test/out/debug/contract_output_test-abi.json",
+        &wallet,
+        TxParameters::default(),
+        Salt::from([1u8; 32]),
```

Comment:
If you don't need a specific salt (looks like you don't), you could just use the default salt. cc @vnepveu @digorithm to confirm

Reply:
Yes you can use the default, since the idea is to just test the different path, it doesn't matter that it's the same salt as the one used in `Contract::deploy`